### PR TITLE
[202012] [cherry-pick] Generate switch level dscp_to_tc_map entry from qos_config template

### PR DESCRIPTION
**Pull request by bingwang-ms**
#### Why I did it
This PR is to cherry-pick https://github.com/Azure/sonic-buildimage/pull/11087 into `202012` branch after resolving conflicts.

This PR is to update template for generating switch level `DSCP_TO_TC_MAP` config.
After PR https://github.com/Azure/sonic-buildimage/pull/10565, there will be two DSCP_TO_TC_MAP

- DSCP_TO_TC_MAP|AZURE is the default map, which is used at port level and switch level
- DSCP_TO_TC_MAP|AZURE_TUNNEL is used to remap the priority of tunnel traffic in dualtor deployment

To address the issue, an entry `PORT_QOS_MAP|global` will be added into `config_db`
```
"PORT_QOS_MAP": {
        "global": {
            "dscp_to_tc_map": "AZURE"
        }
}
```
The entry is generated from `qos_config.j2`.

Related PRs
| Repo  | PR title | State |
| ---------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------- |
| sonic-swss | [\[202012\]Apply DSCP_TO_TC_MAP from PORT_QOS_MAP\|global to switch level ](https://github.com/Azure/sonic-swss/pull/2328) | GitHub issue/pull request detail |
|sonic-utilities | [\[202012\]Update db_migrator to support PORT_QOS_MAP\|global](https://github.com/Azure/sonic-utilities/pull/2215) | GitHub issue/pull request detail |

#### How I did it
Update the template `qos_config.j2`.

#### How to verify it
Verified by vstest.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### Description for the changelog

This PR is to update template for generating switch level `DSCP_TO_TC_MAP` config.

#### Link to config_db schema for YANG module changes

**TODO**

### CODE REVIEW

**Review by prsunny — APPROVED** [review]

File: src/sonic-config-engine/tests/sample_output/py3/qos-arista7800r3-48cq2-lc.json [hunk 1]
```diff
@@ -114,6 +114,9 @@
         }
     },
     "PORT_QOS_MAP": {
+        "global": {
+            "dscp_to_tc_map"  : "AZURE"
+        },
```
Comment (neethajohn) [inline src/sonic-config-engine/tests/sample_output/py3/qos-arista7800r3-48cq2-lc.json:119]: All values in this file need to be updated to use the reference format like "[DSCP_TO_TC_MAP|AZURE]"

**Comment by stephenxs** [comment]
AFAIK, there is no logic to apply DSCP_TO_TC table on the switch on 202012 branch.
do we still need the global DSCP_TO_TC map in 202012?

**Comment by bingwang-ms** [comment]
> AFAIK, there is no logic to apply DSCP_TO_TC table on the switch on 202012 branch. do we still need the global DSCP_TO_TC map in 202012?

Yes, we need to apply DSCP_TO_TC to switch on 202012 branch. I posted a PR for swss change on 202012 branch https://github.com/Azure/sonic-swss/pull/2328

File: src/sonic-config-engine/tests/sample_output/py2/qos-arista7800r3-48cq2-lc.json [hunk 2]
```diff
@@ -114,6 +114,9 @@
         }
     },
     "PORT_QOS_MAP": {
+        "global": {
+            "dscp_to_tc_map"  : "AZURE"
+        },
```
[the remark this answers is not in the document]
Comment (bingwang-ms) [reply src/sonic-config-engine/tests/sample_output/py2/qos-arista7800r3-48cq2-lc.json]: Fixed. Thanks

File: src/sonic-config-engine/tests/sample_output/py3/qos-arista7800r3-48cq2-lc.json [hunk 3]
```diff
@@ -115,7 +115,7 @@
     },
     "PORT_QOS_MAP": {
         "global": {
-            "dscp_to_tc_map"  : "AZURE"
+            "dscp_to_tc_map"  : "[DSCP_TO_TC_MAP|AZURE]"
         },
         "Ethernet0": {
             "dscp_to_tc_map"  : "AZURE",
```
Comment (neethajohn) [inline src/sonic-config-engine/tests/sample_output/py3/qos-arista7800r3-48cq2-lc.json:121]: Please change all the outputs in this file to use the reference format

Comment (bingwang-ms) [reply src/sonic-config-engine/tests/sample_output/py3/qos-arista7800r3-48cq2-lc.json]: Updated. Thanks

**Pull request merged**